From 171c50f6e73bcf5866fd1c47e1a86d5476c355ed Mon Sep 17 00:00:00 2001
From: Manolis Papadakis <manopapad@gmail.com>
Date: Wed, 16 Jul 2014 17:13:10 -0700
Subject: [PATCH] Also accept native types in ?LETs

---
 src/proper_transformer.erl | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/proper_transformer.erl b/src/proper_transformer.erl
index 560ecc5..cbed1ca 100644
--- a/src/proper_transformer.erl
+++ b/src/proper_transformer.erl
@@ -270,6 +270,13 @@ rewrite_expr({call,Line,
     NewRawType = rewrite_type(RawType, ModInfo),
     NewProp = rewrite_expr(Prop, ModInfo),
     {call,Line,FunRef,[NewRawType,NewProp]};
+rewrite_expr({call,Line,
+	      {remote,_,{atom,_,proper_types},{atom,_,bind}} = FunRef,
+	      [RawType,Gen,Flag]}, ModInfo) ->
+    NewRawType = rewrite_type(RawType, ModInfo),
+    NewGen = rewrite_expr(Gen, ModInfo),
+    NewFlag = rewrite_expr(Flag, ModInfo),
+    {call,Line,FunRef,[NewRawType,NewGen,NewFlag]};
 rewrite_expr({call,Line,FunRef,Args}, ModInfo) ->
     NewFunRef = rewrite_expr(FunRef, ModInfo),
     NewArgs = [rewrite_expr(A,ModInfo) || A <- Args],
-- 
1.8.3.2

